opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -1645,6 +1629,21 @@ namespace Opm {
         return *well;
     }
 
+    template<typename TypeTag>
+    bool
+    BlackoilWellModel<TypeTag>::
+    hasWell(const std::string& well_name) const
+    {
+        // finding the iterator of the well in wells_ecl
+        auto well = std::find_if(well_container_.begin(),
+                                     well_container_.end(),
+                                     [&well_name](const WellInterfacePtr& elem)->bool {
+                                         return elem->name() == well_name;
+                                     });
+
+        return (well != well_container_.end());
```

Comment:
Using [`std::any_of()`](https://en.cppreference.com/w/cpp/algorithm/all_any_none_of) this is equivalent to
```C++
return std::any_of(well_container_.begin(), well_container_.end(),
    [&well_name](const WellInterfacePtr& elem) -> bool
{
    return elem->name() == well_name;
});
```

Reply:
Looks better, I will update, but wait until benchmark is done to avoid disruption.